src/anomalib/pipelines/jobs/base.py

```diff
@@ -0,0 +1,87 @@
+"""Job from which all the jobs inherit from."""
+
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+from abc import ABC, abstractmethod
+from collections.abc import Iterator
+from typing import Any
+
+from jsonargparse import ArgumentParser, Namespace
+
+
+class BaseConfigParser(ABC):
+    """Base class for config parsers."""
+
+    def __init__(self, name: str) -> None:
+        self.logger = logging.getLogger(name)
+
+    @abstractmethod
+    def add_arguments(self, parser: ArgumentParser) -> None:
+        """Add arguments to the parser."""
+
+    @abstractmethod
+    def config_iterator(self, args: Namespace) -> Iterator:
+        """Return iterator based on the arguments."""
+
+
+class BaseJob:
```

Comment:
Should inherit from ABC

Reply:
And based on our latest discussion, should be Job :)